src/Helpers/TelemetryHelper.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace Acquia\Cli\Helpers;
+
+use Acquia\Cli\AcquiaCliApplication;
+use Acquia\DrupalEnvironmentDetector\AcquiaDrupalEnvironmentDetector;
+use AcquiaCloudApi\Endpoints\Account;
+use drupol\phposinfo\OsInfo;
+use League\OAuth2\Client\Provider\Exception\IdentityProviderException;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+use Webmozart\KeyValueStore\JsonFileStore;
+use Zumba\Amplitude\Amplitude;
+
+class TelemetryHelper {
+
+  /** @var \Symfony\Component\Console\Output\OutputInterface */
+  private $output;
+
+  /** @var \Symfony\Component\Console\Input\InputInterface */
+  private $input;
+
+  /**
+   * @var \Webmozart\KeyValueStore\JsonFileStore
+   */
+  private $acliDatastore;
+
+  /**
+   * @var \Acquia\Cli\Helpers\ClientService
+   */
+  private $cloudApi;
+
+  /**
+   * @var \Webmozart\KeyValueStore\JsonFileStore
+   */
+  private $cloudDatastore;
+
+  public function __construct(
+    InputInterface $input,
+    OutputInterface $output,
+    ClientService $cloud_api,
+    JsonFileStore $acli_datastore,
+    JsonFileStore $cloud_datastore
+  ) {
+    $this->input = $input;
+    $this->output = $output;
+    $this->cloudApi = $cloud_api;
+    $this->cloudDatastore = $cloud_datastore;
+    $this->acliDatastore = $acli_datastore;
+  }
+
+  /**
+   * Initializes Amplitude.
+   *
+   * @param \Zumba\Amplitude\Amplitude $amplitude
+   *
+   * @throws \Exception
+   */
+  public function initializeAmplitude(Amplitude $amplitude, $app_version): void {
+    $send_telemetry = $this->acliDatastore->get(DataStoreContract::SEND_TELEMETRY);
+    $amplitude->setOptOut(!$send_telemetry);
+
+    if (!$send_telemetry) {
+      return;
+    }
+    try {
+      $amplitude->init('956516c74386447a3148c2cc36013ac3');
+      // Method chaining breaks Prophecy?
+      // @see https://github.com/phpspec/prophecy/issues/25
+      $amplitude->setDeviceId(OsInfo::uuid());
+      $amplitude->setUserProperties($this->getTelemetryUserData($app_version));
+      $amplitude->setUserId($this->getUserId());
+      $amplitude->logQueuedEvents();
+    } catch (IdentityProviderException $e) {
+      // If something is wrong with the Cloud API client, don't bother users.
```

Comment:
I'm on the fence about whether to log an error here.

Reply:
I chose not to log an error because any negative experience with telemetry (even a warning message in logs) could invite people to just stop using it. And if there's really a problem with the Cloud API client, it will probably be revealed in mainline commands that will throw an error anyway.